main.cpp

```diff
@@ -91,15 +91,113 @@ good to go!
  Wait for my code review.
  */
 
+
 #include <iostream>
-int main() 
+#include <math.h> 
+
+struct FloatType
+{
+    float add(float, float);
+    float subtract(float, float);
+    float multiply(float, float);
+    float divide(float, float);
+};
+    
+float FloatType::add(float lhs, float rhs) 
+{
+     return lhs + rhs; 
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;    
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs )
+{
+    if( rhs == 0.f )  
+    { 
+         std::cout << "warning, floating point division by zero returns 'inf' ! " << std::endl;
+        return lhs/rhs;
+    }
+    
+    return lhs / rhs;
+}
+
+struct DoubleType
 {
+    double add(double, double);
+    double subtract(double, double);
+    double multiply(double, double);
+    double divide(double, double);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs; 
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{ 
+    return lhs - rhs; 
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs; 
+}
+double DoubleType::divide(double lhs, double rhs)
+{
+    if( rhs == 0.0 ) std::cout << "warning, floating point division by zero returns 'inf' ! " << std::endl;
+    return lhs / rhs; 
+}
```

Comment:
FIXME: 
you used curly braces with every other `if()` statement in this project
be consistent with your coding style and usage of `{ }`
- [ ] revise: add curly braces to the `if()` block